usr/share/rear/rescue/GNU/Linux/310_network_devices.sh

```diff
@@ -133,6 +173,18 @@ for network_interface in $network_interfaces ; do
     test "$network_interface" && physical_network_interfaces+=" $network_interface"
 done
 
+iplink_has_bridge_rc=
+# Function returning whether 'ip link add name NAME type bridge ...' exists
+function iplink_has_bridge () {
+    [ -n "$iplink_has_bridge_rc" ] && return $iplink_has_bridge_rc
+    if ip link help 2>&1 | grep -qw bridge; then
+        iplink_has_bridge_rc=0
+    else
+        iplink_has_bridge_rc=1
+    fi
+    return $iplink_has_bridge_rc
+}
+
```

Comment:
It seems the iplink_has_bridge_rc variable
is meant as some kind of shortcut for the
iplink_has_bridge function but I fear that may
result wrong (i.e. outdated) return values because
iplink_has_bridge() is called in bridge_handling()
and bridge_handling() is called
for physical_network_interface in $physical_network_interfaces
so that I think for each new physical_network_interface
the iplink_has_bridge_rc shortcut variable may have to be
re-initialized to the empty value.
If my analysis is correct I would suggest to keep the code simple
and do it without sophisticated things that go too easily wrong like
<pre>
function iplink_has_bridge () {
    ip link help 2>&1 | grep -qw bridge && return 0
    return 1
}
</pre>
cf. https://github.com/rear/rear/wiki/Coding-Style
<pre>
Better very simple code than oversophisticated (possibly fragile) constructs.
</pre>
Of course I could misunderstand something here and all is o.k.

Reply:
The iplink_has_bridge_rc is computed once, it's the cached result of the iplink_has_bridge function. It will be computed on first call, then function will return the value everytime.
Since "ip link help" is not supposed to change between calls, this is ok.

Reply:
Now even I see that it is only about 'ip link help'!
I had falsely assumed the iplink_has_bridge function
tests whether or not something (called 'iplink')
is somehow related to (i.e. 'has') a 'bridge'.
I.e. the function name led me on that false track.
Perhaps - if you like - you may rename the function into
something like 'ip_link_command_supports_bridge'
to make it more clear what that function actually does.

Reply:
Hmmm...
Why a function at all when it is only run once?
Why not using a plain varaiable like
<pre>
ip_link_supports_bridge='false'
ip link help 2>&1 | grep -qw bridge && ip_link_supports_bridge='true'
</pre>
at the beginning of 310_network_devices.sh and
later use the ip_link_supports_bridge variable like
<pre>
if is_true $ip_link_supports_bridge ; then
   ...
else
   ...
fi
</pre>

Reply:
sure. it's just because it's a copy/paste from my NewNetworking code :-)